### DISCUSSION TITLE
Restructure to Basic Navigation

### DISCUSSION THREAD

**Discussion by LePips** [jellyfin/Swiftfin#431] [Discussions]
This is just a developer note to outline a restructure the current apps, not necessarily talk about missing functionality (like Search on tvOS).

## General Episode/Item Navigation
At time of writing, in order to get to an episode starting from home (not including Continue Watching or Resume), users must go from `Series -> Season -> Episode -> Hit Play` to start. Movies are fine right now. This behavior is just a result from easy development and to allow viewing _each_ item however is bad for UX as it requires 3 screens (not including home) and 4 taps in order to start playback on an episode.

I plan on reworking it to match what's expected of current apps, where a series is selected and the resume/next up/first episode is chosen to be played from the play button. From there seasons and episodes can be decided further. Essentially, more like the Apple TV app. All information is still accessible but stuff like a season won't be a "main screen" anymore.

On iOS, search will become a separate tab to have 3 tabs: Home, Search, Media (change from All Media to match web and seems grammatically redundant). 

## iPadOS Navigation
To further mirror the Apple TV app and create a better experience for iPadOS, creating the sidebar with the Home, Search, LiveTV, Favorites, _each_ library, and Settings. I will have to see how Stinsen allows this and _may_ have to look at removing landscape mode from iPhones due to this (not including the player). I personally think landscape-only for iPhone is also just generally be more appropriate.

**Comment by LePips** [comment]
@PangMo5 Thoughts or changes?

**Reply by PangMo5** [reply]
I agree with you very much regarding item navigation.

*in docs*
>Stinsen does not support DoubleColumnNavigationViewStyle. The reason for this is that it does not work as expected due to issues with >isActive in SwiftUI. Workaround: Use UIViewRepresentable or create your own implementation.

https://github.com/rundfunk47/stinsen/issues/8#issuecomment-828968779

Officially, stinsen does not appear to support additional navigation styles
As written in the document, it should be bypassed, such as using `UIViewRepresentable`.

**Reply by LePips** [reply]
I saw that, however I also saw somebody create a custom coordinator to support the split view on iPad by practically rewriting TabCoordinatable for it. I'll probably take the same path.

Edit: got something like described above working!
